### PR TITLE
Revert fastpath

### PR DESCRIPTION
**Pull request by jrivera-px**
We had an issue compiling on kernel 5.7.0-1.el8.elrepo.x86_64.   The fix which Prabir had provided for master was based on existing fastpath code.   Since 2.6.0 is not being shipped with fastpath we decided to undo the several fastpath commits and just add the fix for the kernel compilation.

### CODE REVIEW

**Comment by sulakshm** [comment]
This has lots of change, was the revert based on git revert or manually undone?
Can we please have some BVT result before we consider approving.

File: Makefile.in [hunk 1]
```diff
@@ -155,7 +87,7 @@ docker-build: docker-build-dev
 	portworx/px-fuse:dev make
 
 px_version.c:
-	echo "const char *gitversion = \"$(shell git name-rev --name-only HEAD):$(shell git rev-parse HEAD)\";" > $@
+	echo "const char *gitversion = \"$(shell git rev-parse HEAD)\";" > $@
```
Comment (sulakshm) [inline Makefile.in:90]: can we not keep the branch name as well, along with commit id, why this needs to be reverted.

**Comment by jrivera-px** [comment]
Yeah definitely will be doing that before we use this.   I will set up
branch and see how it goes.   Thx

File: pxd.c [hunk 2]
```diff
@@ -99,30 +143,6 @@ static long pxd_ioctl_init(struct file *file, void __user *argp)
 	return pxd_read_init(&ctx->fc, &iter);
 }
 
-static long pxd_ioctl_resize(struct file *file, void __user *argp)
```
Comment (sulakshm) [inline pxd.c:102]: THis change is for supporting fs resize, not fastpath related.

File: pxd.c [hunk 3]
```diff
@@ -166,8 +186,6 @@ static long pxd_control_ioctl(struct file *file, unsigned int cmd, unsigned long
 	case PXD_IOC_INIT:
 		status = pxd_ioctl_init(file, argp);
 		break;
-	case PXD_IOC_RESIZE:
```
Comment (sulakshm) [inline pxd.c:169]: Here too, not related to fastpath

**Comment by jrivera-px** [comment]
Oh and the most of it was a git revert.  Had some issues reverting pxd.c so
I took the version just in the hash before the initial fastpath commit.

File: pxd.c [hunk 4]
```diff
@@ -1470,7 +1071,7 @@ static int pxd_control_open(struct inode *inode, struct file *file)
 		return -EINVAL;
 	}
 
-	cancel_delayed_work_sync(&ctx->abort_work);
+	del_timer_sync(&ctx->timer);
```
Comment (sulakshm) [inline pxd.c:1074]: This is unrelated to fastpath

File: pxd.c [hunk 5]
```diff
@@ -1504,7 +1104,7 @@ static int pxd_control_release(struct inode *inode, struct file *file)
 		pxd_printk("%s: not opened\n", __func__);
 	else
 		ctx->fc.connected = 0;
-	schedule_delayed_work(&ctx->abort_work, pxd_timeout_secs * HZ);
+	mod_timer(&ctx->timer, jiffies + (pxd_timeout_secs * HZ));
```
Comment (sulakshm) [inline pxd.c:1107]: This is unrelated to fastpath

File: pxd.c [hunk 6]
```diff
@@ -1572,14 +1200,18 @@ int pxd_context_init(struct pxd_context *ctx, int i)
 	ctx->miscdev.name = ctx->name;
 	ctx->miscdev.fops = &ctx->fops;
 	INIT_LIST_HEAD(&ctx->pending_requests);
-	INIT_DELAYED_WORK(&ctx->abort_work, pxd_abort_context);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0)
```
Comment (sulakshm) [inline pxd.c:1203]: This was changed to use delayed work for a reason, unrelated to fastpath.

File: pxd.c [hunk 7]
```diff
@@ -1572,14 +1200,18 @@ int pxd_context_init(struct pxd_context *ctx, int i)
...
+	setup_timer(&ctx->timer, pxd_timeout, (unsigned long) ctx);
+#endif
 	return 0;
 }
 
 static void pxd_context_destroy(struct pxd_context *ctx)
 {
 	misc_deregister(&ctx->miscdev);
-	cancel_delayed_work_sync(&ctx->abort_work);
+	del_timer_sync(&ctx->timer);
```
Comment (sulakshm) [inline pxd.c:1214]: not related to fastpath

**Comment by jrivera-px** [comment]
Yeah so thinking about what I had to do with pxd.c is not correct.  I will
look at this again tomorrow cause I believe pxd.c will not be right...

**Comment by sulakshm** [comment]
I will still suggest to fix compilation only instead of reverting changes from way back. The chances of introducing regression pursuing this path is very high in a branch that closer to release.
